### PR TITLE
fix(dtatsample): fix rate infinity cause range error

### PR DESCRIPTION
**Pull request by Cuiyansong**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

## Misc

- [ ] The API has been changed (apache/echarts-doc#xxx).
- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

**Comment by plainheart** [comment]
Why is your chart width zero?

**Comment by Cuiyansong** [comment]
> Why is your chart width zero?

In our application(datart), we set default container with to zero, and wait dataset back and debounce calculate the frame container width and height. For sure, we could set default is 1px or more, then the chart will reset width to 600px for example, the graph will be twinking.  
For sure when the width set to 0, is it not make sense to draw any graph for ECharts, but sometime it is useful to init echart instance and resource, so it is much better to avoid `Exception` I think. In other word, it is seemed not friendly error message when width to zero.

File: src/processor/dataSample.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export default function dataSample(seriesType: string): StageHandler {
                 const size = Math.abs(extent[1] - extent[0]) * (dpr || 1);
                 const rate = Math.round(count / size);
 
-                if (rate > 1) {
```
Comment (pissang) [inline src/processor/dataSample.ts:97]: Using `isFinte(rate)` may be better

Comment (Cuiyansong) [reply src/processor/dataSample.ts:97]: > Using `isFinte(rate)` may be better

Sure, is it better for `if (isFinite(rate) && rate > 1)` ?

Comment (pissang) [reply src/processor/dataSample.ts:97]: Yes I think so

**Comment by pissang** [comment]
Looks good.

**Review by pissang — APPROVED** [review]

**Pull request merged**